llvm/include/llvm/CodeGen/TargetFrameLowering.h

```diff
@@ -286,7 +286,11 @@ class TargetFrameLowering {
   /// hasFP - Return true if the specified function should have a dedicated
   /// frame pointer register. For most targets this is true only if the function
   /// has variable sized allocas or if frame pointer elimination is disabled.
-  virtual bool hasFP(const MachineFunction &MF) const = 0;
+  /// For all targets, this is false if the function has the naked attribute
+  /// since there is no prologue to set up the frame pointer.
+  bool hasFP(const MachineFunction &MF) const {
+    return !MF.getFunction().hasFnAttribute(Attribute::Naked) && hasFPImpl(MF);
```

Comment:
I'm not sure I like forcing it like this; maybe some weird target wants a frame pointer with naked functions?

Would it be much effort to update all targets to check this as a default base implementation? We could also put the other common cases in there like variable frame objects, I guess

Reply:
> If my reading of the GCC attribute documentation is correct, naked functions should behave in the same way as if you had just declared a function in assembly plus the appropriate directives to emit `.cfi_{end,start}proc`, `.size` etc. but no code being generated - so the function can only have a frame pointer if the inline assembly sets one up?

This is my understanding too FWIW.